### ISSUE TITLE
Unused const_auto.go

### ISSUE THREAD

**Issue by nxshock** [kitech/go-toxcore#14]
Looks like there is unused code in `const_auto.go`.

For example, let's see this function: https://github.com/kitech/go-toxcore/blob/c67b0c5f0e937bfd2ccf956ae54afa25e444442c/group.go#L198

This error type is `C.TOX_ERR_CONFERENCE_NEW`: https://github.com/kitech/go-toxcore/blob/c67b0c5f0e937bfd2ccf956ae54afa25e444442c/group.go#L202

On error user just get an error code: https://github.com/kitech/go-toxcore/blob/c67b0c5f0e937bfd2ccf956ae54afa25e444442c/group.go#L205

But we have `const_auto.go`, which can explain this error to user.

---

May be we can fix this? I see following way to solve this problem:
1. Declare error types like this:
```go
var ERR_CONFERENCE_NEW = make(map[int]error)

func init() {
	ERR_CONFERENCE_NEW[0] = nil // no error
	ERR_CONFERENCE_NEW[1] = errors.New("the conference instance failed to initialize")
}
```
2. Use error maps like this:
```go
func (this *Tox) ConferenceNew() (uint32, error) {
	...

	var cerr C.TOX_ERR_CONFERENCE_NEW
	r := C.tox_conference_new(this.toxcore, &cerr)
	if ERR_CONFERENCE_NEW[cerr] != nil { // check for error
		return uint32(r), toxerrf("add group chat failed: %v", ERR_CONFERENCE_NEW[cerr]) // return error description
	}

	...
}
```

**Comment by kitech** [comment]
As you say, it is.
I also want go-toxcore return both error no and error message, so simple parse error message from tox.h's comment, and convert the error info into go source code.
But not update the binding code yet.

**Comment by nxshock** [comment]
Please consider my dirty draft of `go-toxcore/tools/generrmsg.go`:
https://gist.github.com/nxshock/1072d5e4587f7c74f18992128078f300

Result `go-toxcore/const_auto.go`:
https://gist.github.com/nxshock/b89f50b0e1f387c4863b89c60b28814b

Besides some rework I introduce this func to convert C-style error to Go-style error:
```go
func ParseError(group ErrorGroup, code ErrorCode) error
```

Example usage:
```go
func (this *Tox) ConferenceNew() (uint32, error) {
	...
	var cerr ErrorCode
	r := C.tox_conference_new(this.toxcore, &cerr)
	if goErr := ParseError("TOX_ERR_CONFERENCE_NEW", cerr); goErr != nil {
		return uint32(r), goErr
	}
		
	if this.hooks.ConferenceNew != nil {
		this.hooks.ConferenceNew(uint32(r))
	}
	return uint32(r), nil
}
```

**Comment by kitech** [comment]
ErrGroup is nice

**Comment by nxshock** [comment]
Fixed by #18
